src/functions/dialog/index.ts

```diff
@@ -21,11 +47,15 @@ type SpawnDialogOptions = {
  * @param options - Spawning options
  * @return Promise resolved with the `close` event payload
  */
-export function spawnDialog(
-	dialog: Component,
-	props: object = {},
+export function spawnDialog<
+	C extends Component,
+	Props extends Record<EventPropName<'close'>, (...payload: unknown[]) => unknown> = ComponentProps<C>, // eslint-disable-line space-before-function-paren
+	Payload extends unknown[] = EventPayload<NonNullable<Props[EventPropName<'close'>]>>,
+>(
+	dialog: C,
```

Comment:
I don't understand why it works, we `C` is specified as `typeof MyDialog`, but it cannot infer it from `MyDialog` passed to the `dialog` param...